pilot/pkg/bootstrap/server.go

```diff
@@ -535,16 +536,26 @@ func (s *Server) initDNSTLSListener(dns string) error {
 	}
 	certDir := dnsCertDir
 
-	key := path.Join(certDir, constants.KeyFilename)
-	cert := path.Join(certDir, constants.CertChainFilename)
+	key := model.GetOrDefault(defaultTLSServerKey, path.Join(certDir, constants.KeyFilename))
+	cert := model.GetOrDefault(defaultTLSServerCertChain, path.Join(certDir, constants.CertChainFilename))
 
 	certP, err := tls.LoadX509KeyPair(cert, key)
 	if err != nil {
 		return err
 	}
 
 	cp := x509.NewCertPool()
-	rootCertBytes := s.ca.GetCAKeyCertBundle().GetRootCertPem()
+	var rootCertBytes []byte
+	var defaultRootCertBytes []byte
+	if defaultTLSServerRootCert != "" {
+		defaultRootCertBytes, err = ioutil.ReadFile(defaultTLSServerRootCert)
+	}
+
+	if err == nil && defaultRootCertBytes != nil {
```

Comment:
we should handle the error properly, here we silently ignore it

Reply:
I will add a handler for it. If the value is set and we are unable to read the cert then it should error out.